### DISCUSSION TITLE
Make DataConnection.Disposed or CheckAndThrowOnDisposed() public?

### DISCUSSION THREAD

**Discussion by Drapondur** [linq2db/linq2db#4303] [Ideas]
Would you like to consider to make DataConnection.Disposed  public? Or CheckAndThrowOnDisposed()?

The reason for my request is, that I implemented extension methods for DataConnection and I want to make sure that they check if the context has been disposed. Amongst other things I am holding a reference to a class instance and to the DataConnection. I am also adding an DataContextInterceptor in order to free resources when the Connection has been closed. Unfortunately AddInterceptor does not throw an ObjectDisposedException.

However, if my extension method is called on a disposed DataConnection my resources never get disposed or GC'ed. Therefore I want to throw an ObjectDisposedException if the DataConnection has already become invalid.

As a workaround I could do something like DataConnection.CreateCommand() in order to enforce a check for disposal but that seems like quite some overhead to me.

**Comment by MaceWindu** [comment]
Because `Disposed` property defined as protected, so you can define your connection class inherited from `DataConnection` and access Disposed property from it.
Will this work for you?

**Reply by Drapondur** [reply]
Sorry for not responding for so long. I replied to the mail notification because I assumed that it will show up there. My response was:

Thanks for responding. Unfortunately it won't because it's an extension method, like

public static void MethodName(this DataConnection conn)

For any such extension it might be desirable to be able to check for Disposed just as one ought to check for the parameter not to be null.

Maybe defining it on IDataContext and implementing it explicitly might be a compromise, if you don't want to expose it for the reason of keeping the number of public members of DataConnection low?

Anyway, from my side there's no rush or urgency as it's just me using the code and I should be aware of what I  am doing.

**Reply by MaceWindu** [reply]
We would prefer to avoid such implementation details exposed to public surface.

We should add dispose checks to public context APIs where it is missing (like `AddInterceptor` in your case). If I understand correctly it will solve your issue?

**Reply by MaceWindu** [reply]
#4320

**Reply by Drapondur** [reply]
That'll do it. Thanks.
